client/app/bundles/course/survey/pages/SurveyIndex/__test__/NewSurveyButton.test.jsx

```diff
@@ -0,0 +1,45 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import { mount } from 'enzyme';
+import ReactTestUtils from 'react-addons-test-utils';
+import CourseAPI from 'api/course';
+import injectTapEventPlugin from 'react-tap-event-plugin';
+import storeCreator from '../../../store';
+import NewSurveyButton from '../NewSurveyButton';
+import SurveyFormDialogue from '../../../containers/SurveyFormDialogue';
+
+injectTapEventPlugin();
+
+describe('<NewSurveyButton />', () => {
+  it('injects handlers that allow surveys to be created', () => {
+    const spyCreate = jest.spyOn(CourseAPI.survey.surveys, 'create');
+    const store = storeCreator({ surveys: { surveysFlags: { canCreate: true } } });
+    const contextOptions = {
+      context: { intl, store, muiTheme },
+      childContextTypes: { muiTheme: React.PropTypes.object, intl: intlShape },
+    };
+
+    const newSurveyButton = mount(<NewSurveyButton />, contextOptions);
+    const surveyFormDialogue = mount(<SurveyFormDialogue />, contextOptions);
+
+    // Click 'new survey' button
+    const newSurveyButtonNode = ReactDOM.findDOMNode(newSurveyButton.find('button').node);
+    ReactTestUtils.Simulate.touchTap(newSurveyButtonNode);
+    expect(surveyFormDialogue.find('SurveyFormDialogue').first().props().visible).toBe(true);
+
+    // Enter survey title
+    const surveyTitle = 'Funky survey title';
+    const surveyForm = document.body.querySelector('form');
+    const titleInput = surveyForm.querySelector('input[name="title"]');
+    titleInput.value = surveyTitle;
+    ReactTestUtils.Simulate.change(titleInput);
+
+    // Submit survey form
+    const submitButton = surveyFormDialogue.find('FormDialogue').first().node.submitButton;
+    ReactTestUtils.Simulate.touchTap(ReactDOM.findDOMNode(submitButton));
+
+    expect(spyCreate).toHaveBeenCalled();
```

Comment:
We should test the exact arguments, this will guarantee the args send to the server is correct.
```
expect(spyCreate).toHaveBeenCalledWith(arguments)
```